### PR TITLE
Fix action deletion/addition when more than one interface are involved

### PR DESCRIPTION
**Pull request by sethalves**
- remember uuid of deleted actions and don't allow them to be readded
- handle adding an action while also getting updates about the entity.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: fa7a5f1a615e599f6f7e7e8cbe9878a67b5c464c available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -1668,10 +1674,20 @@ void EntityItem::deserializeActionsInternal() {
...
     }
 
+    // trim down _previouslyDeletedActions
+    QMutableHashIterator<QUuid, quint64> _previouslyDeletedIter(_previouslyDeletedActions);
+    while (_previouslyDeletedIter.hasNext()) {
+        _previouslyDeletedIter.next();
+        if (now - _previouslyDeletedIter.value() > _rememberDeletedActionTime) {
+            _previouslyDeletedActions.remove(_previouslyDeletedIter.key());
+        }
+    }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: _rememberDeletedActionTime is in units of **seconds** according to comment higher up, but the math comparison here appears to be using microseconds.  Something is wrong.

Comment (sethalves) [inline libraries/entities/src/EntityItem.cpp]: fixed.  (i hope)

**Comment by hifi-gustavo** [comment]
Build console output for commit: f2ebba9863a077cff9a008eb98082e470be41c79 available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e90b156b7bd7c8456e6a6376cdf0db8f861a6c2f available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 40c5103d2fba7b3cbd2cc047cd002086651d6a1c available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9f27e5ff1fb260bc313fd8c9c08775ab8eae47f9 available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bb5abeeabbb23b50b2c791ef330100c380714bb3 available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5b970e6b1af1a3d06efd73ce868e53d0d65ea96e available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 880d92ee64d6cd369731c9b2449df4ab6ee214fd available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c0e8b02a2fddc5768c1b6bc97d225523f47ce44b available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 67cc944afc27ce9182b716a9b7c4ab07547901e6 available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a737b4024d395921609c5e499819bc8c33f324a4 available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 993111d22f6249b788c4ca37f51dce9ca85d63ee available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Comment by AndrewMeadows** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 993111d22f6249b788c4ca37f51dce9ca85d63ee available at: https://gist.github.com/657296f5d56a062ffb8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5976.dmg

**Pull request merged**